src/PcodeFixupPreprocessor.cpp

```diff
@@ -0,0 +1,34 @@
+// SPDX-FileCopyrightText: 2024 Crabtux <crabtux@mail.ustc.edu.cn>
+// SPDX-License-Identifier: LGPL-3.0-or-later
+
+#include "PcodeFixupPreprocessor.h"
+
+#include <funcdata.hh>
+#include <flow.hh>
+#include <override.hh>
+
+using namespace ghidra;
+
+void PcodeFixupPreprocessor::fixupSharedReturnJumpToRelocs(RzAnalysisFunction *function, Funcdata *func, RzCore *core, RizinArchitecture &arch)
+{
+	RzList *refs = rz_analysis_function_get_xrefs_from(function);
+
+	RzListIter *it;
+	RzAnalysisXRef *xref;
+
+	// C++ has more strict type checking than C, which stops us from implicitly casting void * to RzListIter *.
+	// Expand the rz_list_foreach macro manually. 
+	if (refs) for (it = refs->head; it && (xref = (RzAnalysisXRef *)it->elem, 1); it = it->next)
```

Comment:
potentially you could reuse (and rename) https://github.com/rizinorg/cutter/blob/dev/src/core/RizinCpp.h#L110 or https://github.com/rizinorg/cutter/blob/dev/src/core/RizinCpp.h#L56

Reply:
It seems that rz-ghidra has already implemented `rz_list_foreach_cpp` in [RizinUtils.h](https://github.com/rizinorg/rz-ghidra/blob/dev/src/RizinUtils.h). Maybe I can rewrite my code to use that :)